pkg/controller/statusmanager/pod_status.go

```diff
@@ -84,53 +84,93 @@ func SetNodeCondition(conditions *[]corev1.NodeCondition, newCondition corev1.No
 		*conditions = append(*conditions, newCondition)
 		return true
 	}
-
-	if existingCondition.Status != newCondition.Status {
+	if (existingCondition.Status != newCondition.Status || existingCondition.Reason != newCondition.Reason ||
+		existingCondition.Message != newCondition.Message) {
+		changed = true
 		existingCondition.Status = newCondition.Status
+		existingCondition.Reason = newCondition.Reason
+		existingCondition.Message = newCondition.Message
 		existingCondition.LastTransitionTime = metav1.NewTime(time.Now())
-		changed = true
 	}
-
-	existingCondition.Reason = newCondition.Reason
-	existingCondition.Message = newCondition.Message
 	return changed
 }
 
-func (status *StatusManager) setNodeNetworkUnavailable(nodeName string, ready bool, reason string, message string, sharedInfo *sharedinfo.SharedInfo) {
+// CheckExistingAgentPods is for a case: nsx-node-agent becomes unhealthy -> NetworkUnavailable=True -> operator off ->
+// nsx-node-agent becomes healthy and keeps running -> operator up -> operator cannot receive nsx-node-agent event to
+// set NetworkUnavailable=False. So a full sync at the start time is necessary.
+func (status *StatusManager) CheckExistingAgentPods (firstBoot *bool, sharedInfo *sharedinfo.SharedInfo) error {
+	status.Lock()
+	defer status.Unlock()
+
+	if !*firstBoot {
+		return nil
+	}
+	log.Info("Checking all nsx-node-agent pods for node condition")
+	pods := &corev1.PodList{}
+	err := status.client.List(context.TODO(), pods, client.MatchingLabels{"component": operatortypes.NsxNodeAgentContainerName})
+	if err != nil {
+		log.Error(err, "Error getting pods for node condition")
+		return err
+	}
+	if len(pods.Items) == 0 {
+		log.Info("nsx-node-agent not found for node condition")
+		nodes := corev1.NodeList{}
+		err = status.client.List(context.TODO(), &nodes)
+		if err != nil {
+			log.Error(err, "Failed to get nodes for condition updating")
+			return err
+		}
+		for _, node := range nodes.Items {
+			status.setNodeNetworkUnavailable(node.ObjectMeta.Name, false, nil, "NSXNodeAgent", "Waiting for nsx-node-agent to be created", sharedInfo)
+		}
+		*firstBoot = false
+		return nil
+	}
+	podName := types.NamespacedName{}
+	for _, pod := range pods.Items {
+		status.setNodeConditionFromPod(podName, sharedInfo, &pod)
+	}
+	*firstBoot = false
+	return nil
+}
+
+func (status *StatusManager) setNodeNetworkUnavailable(nodeName string, ready bool, startedAt *time.Time, reason string, message string, sharedInfo *sharedinfo.SharedInfo) {
 	if sharedInfo.LastNetworkAvailable == nil {
 		sharedInfo.LastNetworkAvailable = make(map[string]time.Time)
 	}
+	if ready {
+		log.V(1).Info(fmt.Sprintf("Last network unavailable time of node %s is set to %s", nodeName, *startedAt))
```

Comment:
What is log level `V(1)` is? previously I thought the less number is the higher log level

Reply:
https://github.com/go-logr/logr#why-v-levels
V(1) equals to DEBUG, in my test:
`2021-01-14T05:46:07.420Z        DEBUG   status_manager  Last network unavailable time of node wdc-rdops-vm09-dhcp-224-189 is set to 2021-01-14 05:45:49 +0000 UTC`

Reply:
Thanks!